pkg/controller/common/reconcile.go

```diff
@@ -0,0 +1,180 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+
+	ign3types "github.com/coreos/ignition/v2/config/v3_4/types"
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	"k8s.io/klog/v2"
+)
+
+// IsRenederedConfigReconcilable checks the new rendered config against the old
```

Comment:
Note to reviewer: I created two functions to replace the `reconcile()` function in the MCD. This first one is used against rendered MachineConfigs. It just calls to the second one. The second function accepts a list of component MachineConfigs so that we can identify which added MachineConfig is unreconcilable. That information is included in the returned error.